implementation/src/main/java/io/smallrye/config/ConfigMappingProvider.java

```diff
@@ -649,14 +649,40 @@ private void mapConfiguration(SmallRyeConfig config, ConfigMappings mappings) th
         mappings.registerConfigMappings(context.getRootsMap());
     }
 
-    private boolean isPropertyInRoot(String propertyName) {
+    private boolean isPropertyInRoot(NameIterator propertyName) {
```

Comment:
In the previous version, `isPropertyInRoot` returns true if *any* registered root contains the property.
But in this version, it returns true only when *all* registered roots contain the property.

Is this behavior change intentional?

Reply:
Probably not :)

Reply:
I guess I've messed up on the branching and pushed and old version when I've renamed `master` to `main`. I took the chance and added a few more tests. Do you mind reviewing it? :)